tests/sqlalchemy_dataset/test_sqlalchemydataset.py

```diff
@@ -100,13 +100,14 @@ def test_only_connection_string(sa):
     SqlAlchemyDataset('test_engine', connection_string='sqlite://')
 
 
-def test_schema_custom_sql_error(sa):
-    engine = sa.create_engine('sqlite://')
-
-    with pytest.raises(ValueError) as err:
-        SqlAlchemyDataset('test_schema_custom', schema='example', engine=engine,
-                          custom_sql='SELECT * FROM example.fake')
-        assert "Cannot specify both schema and custom_sql." in str(err.value)
+# NOTE: 20200306: The behavior of allowing both custom_sql and schema is currently expected
```

Comment:
agreed... so why not just delete all this code?

Reply:
When you say it like that....

I was viewing it a message to our future selves for when we try to really figure out how to best handle schemas. But I think your approach is better, and what it really suggests is I should do some synthesis first. I propose we add the below into a comment in the SqlAlchemyDataset class.

Several cases we need to consider:

1. Distributed backends (e.g. Snowflake and BigQuery) often use a <database>.<schema>.<table> syntax, but currently we are biased towards only allowing schema.table

2. In the wild, we see people using several ways to declare the schema they want to use:
  a. In the connection string, the original RFC only specifies database, but schema is supported by some backends (Snowflake) as a query parameter.
  b. As a default for a user (the equivalent of USE SCHEMA being provided at the beginning of a session)
  c. As part of individual queries.

3. We currently don't make it possible to select from a table in one query, but create a temporary table in another schema, except for with BigQuery and (now) snowflake, where you can specify the table name (and potentially trip of database, schema, table) in the batch_kwargs.

4. The SqlAlchemyDataset interface essentially predates the batch_kwargs concept and so part of what's going on, I think, is a mismatch between those. I think we should rename custom_sql -> "temp_table_query" or similar, for example.

Reply:
I added a comment to this effect in the `create_temporary_table` method.